File: 02-js-moderno/presupuestoTotal.js
const carrito = [
    { nombre: "Monitor", precio: 300 },
    { nombre: "Teclado"},
    { nombre: "Ratón", precio: 20 }
];

const precioTotal = carrito.reduce((total, producto) => (producto.precio || 0) + total, 0)
console.log(precioTotal);

File: 01-java-moderno/PresupuestoTotal.java

void main() {
    record Carro(String producto, Integer precio){}

    List<Carro> productos = List.of(
            new Carro("Monitor", 300),
            new Carro("Teclado", 60),
            new Carro("Ratón", 40),
            new Carro("Auriculares", null)
    );

    double precioCarrito = productos
            .stream()
            //.mapToDouble(p -> p.precio)
             .mapToDouble(p -> Optional.ofNullable(p.precio).orElse(0))
            .sum();
    System.out.println(precioCarrito);
}